prime-router/metadata/fhir_mapping/hl7/resource/MessageHeader.yml

```diff
@@ -78,3 +76,17 @@ extension_3:
         system: "http://terminology.hl7.org/CodeSystem/v3-AcknowledgementCondition"
       vars:
         code: MSH.16
+
+extension_4:
+  generateList: true
+  expressionType: nested
+  expressionsMap:
+    url:
+      type: SYSTEM_URL
+      value: message-sent
+    valueDateTime:
+      type: STRING
+      valueOf: "GeneralUtils.dateTimeWithZoneId(dateTimeIn,ZONEID)"
+      expressionType: JEXL
+      vars:
+        dateTimeIn: MSH.7
```

Comment:
We are already storing the MSH.7 inside Bundle.timestamp. That logic is inside `HL7toFhirTranslator.enhanceBundleMetadata`. We need to decide if we need to keep that logic or keep this extension and then update the FHIR to HL7 mapping to grab the value from either the extension or  Bundle.timestamp. Also, I don't see why we can't have a mapping to  set Bundle.timestamp? Maybe because it's not under a resource

Reply:
I added this because I was not seeing it as a dateTime in the message. It looks like it is for some reason being put as a string for a visit number in the identifier for the ServiceRequest? That doesn't seem right to me. 

{
        "type" : {
          "coding" : [ {
            "system" : "http://terminology.hl7.org/CodeSystem/v2-0203",
            "code" : "VN",
            "display" : "Visit number"
          } ]
        },
        "value" : "20230506052916-0500"
      }

Reply:
Discussed and will remove. This is getting sent over it just did not look like it via search because it was being converted into another timezone.